src/components/AddTodoForm/AddTodoForm.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import users from '../../api/users';
+
+import { getUsersById } from '../../utils/getUsersById';
+import { getLargestId } from '../../utils/getLargestId';
+import { Props } from '../../types/Prop';
+
+export const AddTodoForm: React.FC<Props> = (props: Props) => {
+  const {
+    setTodos,
+    title,
+    setTitle,
+    titleError,
+    setTitleError,
+    selectedUserId,
+    setSelectedUserId,
+    userIdError,
+    setUserIdError,
+  } = props;
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(
+      event.target.value
+        .trimStart()
+        .replace(/[^a-zA-Zа-щА-ЩЬьЮюЯяЇїІіЄєҐґ0-9 ]/g, ''),
+    );
+    setTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const handleReset = () => {
+    setTitle('');
+    setTitleError(false);
+    setSelectedUserId(0);
+    setUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!selectedUserId) {
+      setUserIdError(true);
+    }
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    setTodos(currentTodos => [
+      ...currentTodos,
+      {
+        id: getLargestId(currentTodos),
+        title,
+        completed: false,
+        userId: selectedUserId,
+        user: getUsersById(selectedUserId),
```

Comment:
Ensure that the `getUsersById` function correctly returns a user object. If it returns undefined or null, this could lead to issues when rendering the user information in the todo list.